### DISCUSSION TITLE
Multipart responses

### DISCUSSION THREAD

**Discussion by ghost** [hyperium/hyper#2591] [Q&A] [answered]
**Version**
0.14

**Platform**
Linux nixos 5.11.21 #1-NixOS SMP Fri May 14 08:50:20 UTC 2021 x86_64 GNU/Linux

**Description**
I want to know how to send a multipart response from a server built using hyper. I'm trying to add support for `Range` header in a server:

https://developer.mozilla.org/en-US/docs/Web/HTTP/Range_requests

**Comment by seanmonstar** [comment, marked as answer]
hyper doesn't do anything special to implement range requests. You could use a server framework that provides support, like [warp](https://crates.io/crates/warp). Or you can just handle them yourself with hyper. That'd mean:

- Sending an `Accept-Range: bytes` header in responses that support it.
- Parsing a `Range` header in requests.
- Using those parsed indices, return a body with just the parts requested.
